### ISSUE TITLE
Readme Instructions do not appear to work as described

### ISSUE THREAD

**Issue by BryanCrotaz** [ember-cli/broccoli-caching-writer#70]
Example: https://github.com/silver-curve/broccoli-archiver

`build` function creates a zip file from the input tree

**Expected** zip file is created in both output folder and cache folder
**Actual** zip file is created in output folder only. Cache folder is empty

Adding the following options makes no difference:

```
	options.cacheInclude = [/.*/];
	options.inputFiles = ['**/*'];
```

**Comment by BryanCrotaz** [comment]
Simplest possible BrocFile:
```
var archiver = require('broccoli-archiver');

module.exports = new archiver("src");
```

This shows `broccoli-archiver`'s build method being called every time even when no source files have changed.

**Comment by BryanCrotaz** [comment]
@joliss is this something you can comment on? Feels like I've built a very simple plugin but I can't for the life of me work out how to enable caching

**Comment by BryanCrotaz** [comment]
for avoidance of doubt I'm running with broccoli-clext
`broccoli-clext build dist --once`

**Comment by BryanCrotaz** [comment]
Seems like caching only works if you leave broccoli running but if you run broccoli twice the cache doesn't work - build gets called twice.

https://github.com/broccolijs/broccoli-plugin/issues/11

**Comment by stefanpenner** [comment]
> Seems like caching only works if you leave broccoli running but if you run broccoli twice the cache doesn't work - build gets called twice.

his is by design. broccoli-persistent-filter does have a second level of caching, which allows it to persist beyond reloads. But it is very tricky to get the cache keys just right, and with persistent-filter the inputs aren't opaque but with this tool, they are quite tricky.

TL;DR that sort of caching, done without issue will most likely (for now anyways) be a one-off solution for the specific use-case.

**Comment by BryanCrotaz** [comment]
Where are the caveats documented?

**Comment by stefanpenner** [comment]
@BryanCrotaz typical cache key problems, if something isn't included in the cache key, and it and it alone changes you will have a stale build.

**Comment by BryanCrotaz** [comment]
What needs to go in the cache key other than the last modified of the source files?

**Comment by stefanpenner** [comment]
@BryanCrotaz what other information may change between restarts?

The source-code that is being executed (and their dependent files), the node version, the OS, the ENV vars etc.etc.
